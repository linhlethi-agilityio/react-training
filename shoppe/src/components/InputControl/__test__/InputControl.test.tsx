import { RenderResult, fireEvent, render } from '@testing-library/react';
import { StyledInputControl } from '..';

describe('testing InputControl component', () => {
  let wrapper: RenderResult;

  const onChange = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <StyledInputControl label="Name" value="value" name="name" id="name" onChange={onChange} />,
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = wrapper;

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render InputControl correctly', () => {
    const { getByLabelText } = wrapper;

    const input = getByLabelText('Name:') as HTMLInputElement;

    expect(input.value).toBe('value');
    expect(input.name).toBe('name');
  });

  test('should call onChange when change input', () => {
    const { getByLabelText } = wrapper;

    const input = getByLabelText('Name:') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'new value' } });
    expect(onChange).toHaveBeenCalled();
  });
});
