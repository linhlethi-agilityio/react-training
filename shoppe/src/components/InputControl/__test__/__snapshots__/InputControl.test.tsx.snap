// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing InputControl component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gueYoa DaLyK"
  >
    <label
      for="name"
    >
      Name:
    </label>
    <input
      class="sc-beqWaB iqNjic"
      id="name"
      name="name"
      value="value"
    />
  </div>
</DocumentFragment>
`;
